R/parsing_partitions.R

```diff
@@ -0,0 +1,62 @@
+setwd('/home/striker/Desktop/UCLA/BS Research Paper/SideWork')
+
+#As a start, try turning the bit that parses the .partitions file into a function 
+#that returns a data frame with three columns, 
+
+#the first being the gene name, 
+#the second being the starting offset, 
+#and the third being the ending offset.
+
+parsing_partitions <- function(file_name) {
+  #Need to return a dataframe with 3 columns:
+  #1st column: The gene name
+  gene_names <- c()
+  #2nd column: The starting offset
+  starting_offset <- c()
+  #3rd column: The ending offset
+  ending_offset <- c()
+  
+  
+  fileName <- file_name
+  partions_strings <- readChar(fileName, file.info(fileName)$size)
+  
+  partion_strings_newline_split <- scan(text = partions_strings, what = "\n")
+  print(partion_strings_newline_split)
+  
+
+  gene_places <- c()
+  
+  n = length(partion_strings_newline_split)
+  #print(n)
+  for (i in 1:n) {
+    #print(i)
+    #print(partion_strings_newline_split[i])
+    if(partion_strings_newline_split[i] == "DNA,"){
```

Comment:
This is a bit fragile. I think you would probably be better off with using `readLines` (instead of `scan`) and capturing the relevant text from each line using https://stringr.tidyverse.org/reference/str_match.html. Let me know if you need help writing this regular expression. The advantage of using `str_match` is that is vectorized and you therefore don't need to write a for loop, making your code much more concise.